### PR TITLE
Restructuring - Itunes Provider with ItuneDetail and Itune Container

### PR DESCRIPTION
**Pull request by naeemdadi1**
### Ticket Link

---

### Related Links

---

### Description

- Restructured folder for details page.
- Created a new folder ItunesProvider and moved itunesContainer and ituneDetailContainer in it.
- Moved selctor, saga, reducer one level up in ItunesProvider.
- Wrote Reducer, Saga, Selector and ItuneDetailContainer.

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's
feat:itunesProvider

### CODE REVIEW

File: app/components/ItunesCard/index.js [hunk 1]
```diff
@@ -6,6 +6,8 @@ import { truncate } from 'lodash';
 import styled from 'styled-components';
 import { T } from '@components/T';
 import If from '../If/index';
+import history from '@app/utils/history';
+// import { Link } from 'react-router-dom';
```
Comment (alichherawalla) [inline app/components/ItunesCard/index.js]: remove commented out code

**Review by rohan-saroha — CHANGES_REQUESTED** [review]
remove the comment out code know

File: app/components/ItunesCard/index.js [hunk 2]
```diff
@@ -95,6 +97,7 @@ const ItunesCard = ({ itune, handleOnActionClick, intl }) => {
 
   return (
     <CustomCard
+      extra={<Link to={`/track/${trackId}`}>More</Link>}
```
Comment (alichherawalla) [inline app/components/ItunesCard/index.js]: please use the routeConstants

File: app/components/ItunesCard/tests/__snapshots__/index.test.js.snap [hunk 3]
```diff
@@ -18,6 +18,15 @@ exports[`<ItunesCard /> should render and match the snapshot 1`] = `
           >
             Doin' What She Likes
           </div>
+          <div
+            class="ant-card-extra"
+          >
+            <a
+              href="/track/undefined"
```
Comment (alichherawalla) [inline app/components/ItunesCard/tests/__snapshots__/index.test.js.snap]: add valid props for the snapshot

**Review by alichherawalla — APPROVED** [review]

File: app/utils/routeConstants.js [hunk 4]
```diff
@@ -1,10 +1,13 @@
 export default {
-  repos: {
+  itune: {
```
Comment (alichherawalla) [inline app/utils/routeConstants.js:2]: itunes

File: app/utils/routeConstants.js [hunk 5]
```diff
@@ -1,10 +1,13 @@
...
     route: '/',
     props: {
       maxwidth: 500,
       padding: 20
     },
     exact: true
+  },
+  ituneDetail: {
```
Comment (alichherawalla) [inline app/utils/routeConstants.js]: itunesDetails

File: app/utils/routeConstants.js [hunk 6]
```diff
@@ -1,10 +1,13 @@
...
+    route: '/track/:id'
```
Comment (alichherawalla) [inline app/utils/routeConstants.js]: tracks

**Pull request merged**